src/snapred/meta/pointer.py

```diff
@@ -0,0 +1,22 @@
+import ctypes
+from typing import Any
+
+OBJCACHE = {}
+
+
+def create_pointer(thing: Any) -> int:
+    """
+    Returns a memory address to be used as a pointer, and retains a handle to the object
+    to ensure the memory can be accessed later.
+    @param the object whose pointer is required
+    @return the pointer to the object
+    """
+    OBJCACHE[id(thing)] = thing
```

Comment:
so this is mostly so it doenst get garbage collected?

Reply:
correct, without this there are often segfaults